subsys/usb/class/cdc_acm.c

```diff
@@ -1067,7 +1067,7 @@ static const struct uart_driver_api cdc_acm_driver_api = {
 			    CONFIG_USB_CDC_ACM_DEVICE_NAME "_" #x,	\
 			    &cdc_acm_init, &cdc_acm_dev_data_##x,	\
 			    &cdc_acm_config_##x,			\
-			    APPLICATION,				\
+			    POST_KERNEL,				\
```

Comment:
This should be separate patch.

Reply:
Better separate PR, and probably not necessary, why you can not initialize your application at APPLICATION level?

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/23287

It's not the application, it is the OpenThread networking stack - which gets initialized earlier than the application. The CDC_ACM UART is required to be ready by the time the OpenThread-NCP starts since it immediately uses the UART to inform the Linux OpenThread tools about a (re-)start.

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/23287

Reply:
Rebased onto master after https://github.com/zephyrproject-rtos/zephyr/pull/23287 has been merged.